ush/rocoto/workflow_utils.py

```diff
@@ -201,8 +204,11 @@ def create_wf_task(task, cdump='gdas', cycledef=None, envar=None, dependency=Non
     # Add PARTITION_BATCH to all non-service jobs on Orion (SLURM)
     if get_scheduler(detectMachine()) in ['slurm'] and detectMachine() in ['ORION']:
         task_dict['partition'] = '&PARTITION_BATCH;'
-    # Add PARTITION_SERVICE to all service jobs (SLURM)
-    if get_scheduler(detectMachine()) in ['slurm'] and task in ['getic','arch','earc']:
+    # Add PARTITION_SERVICE to all service jobs on Hera (SLURM)
+    if get_scheduler(detectMachine()) in ['slurm'] and task in ['getic','arch','earc'] and detectMachine() in ['HERA','JET']:
+        task_dict['partition'] = f'&PARTITION_{task.upper()}_{cdump.upper()};'
+    # Add PARTITION_BATCH to all fcst jobs on JET (SLURM)
+    if get_scheduler(detectMachine()) in ['slurm'] and task in ['fcst'] and detectMachine() in ['JET']:
```

Comment:
I think I asked this before, why is JET requiring special treatment for `fcst` and not for `post` or other jobs?

Reply:
I believe it is because fcst job can run in different configuration to the best of using the JET partition. I did not recall you asked this question before.

Reply:
Is the above two changes complete your request? Thanks

Reply:
> I'd also like to understand why Jet is a special case here.

Answered.